Unable to compile, missing declarations

**Describe the bug**
I am compiling a program that uses Adafruit_ImageReader. The latest version requires Adafruit_SPIFlash. When compiling using Adafruit_SPIFlash version 3.2.2, everything compiles properly with no errors or warnings. When I upgrade to Adafruit_SPIFlash version 3.3.0, I receive many errors that state values are not declared in the scope (see below for details). When I revert back to 3.2.2, everything compiles correctly.  I wasx able to reproduce this in a very simple program (attached below).

**Set up (mandatory)**
 - **PC & IDE** : Arduino 1.8.12  Windows 10
 - **Board** : NodeMCU-32S
- **Libraries versions**
Adafruit BUSIO 1.50
Adafruit  EPD 2.53
Adafruit GFX 1.10.1
Adafruit ILI9341 1.56
Adafruit ImageReader 2.4.0
Adafruit SPIFlash 3.3.0
SdFat 1.1.4   <- with required changes 
                    //  SdFat library, editing the default SdFatConfig.h file
                   //       #define USE_STANDARD_SPI_LIBRARY 2           
                   //      #define FAT12_SUPPORT 1                       
                   //      #define ENABLE_EXTENDED_TRANSFER_CLASS 1     


 **To Reproduce**
Steps to reproduce the behavior:
Run the following code using the board NodeMCU-32S

 - Program starts here -
#include <SdFat.h>
#include <SPI.h>
#include <Adafruit_GFX.h>       
#include <Adafruit_ILI9341.h> 
#include <Adafruit_ImageReader.h>
#include <Adafruit_ImageReader_EPD.h>
  Adafruit_ImageReader *reader;
  ImageReturnCode ImageReaderStat;
void setup() {
}
void loop() {
}
- Program ends here -


**Serial Log**
...\Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h:61:50: error: 'MSBFIRST' was not declared in this scope

     _spi->beginTransaction(SPISettings(clock_hz, MSBFIRST, SPI_MODE0));
Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h:62:23: error: 'LOW' was not declared in this scope

     digitalWrite(_ss, LOW);

                       ^
...\Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h:62:26: error: 'digitalWrite' was not declared in this scope

     digitalWrite(_ss, LOW);

                          ^
...\Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h: In member function 'void Adafruit_FlashTransport_SPI::endTransaction()':

...\Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h:66:23: error: 'HIGH' was not declared in this scope

     digitalWrite(_ss, HIGH);

                       ^
...\Arduino\libraries\Adafruit_SPIFlash\src/spi/Adafruit_FlashTransport_SPI.h:66:27: error: 'digitalWrite' was not declared in this scope

     digitalWrite(_ss, HIGH);

                           ^
...\Arduino\libraries\Adafruit_SPIFlash\src/Adafruit_SPIFlashBase.h:89:44: error: 'HIGH' was not declared in this scope

       digitalWrite(_ind_pin, _ind_active ? HIGH : LOW);

                                            ^
and many more ...........

--- a/.github/workflows/githubci.yml
+++ b/.github/workflows/githubci.yml
@@ -7,7 +7,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arduino-platform: ['metro_m0', 'metro_m4', 'nrf52840', 'cpx_ada', 'cpb']
+        arduino-platform: ['metro_m0', 'metro_m4', 'nrf52840', 'esp8266', 'esp32']
 
     runs-on: ubuntu-latest
     steps:

--- a/.github/workflows/githubci.yml
+++ b/.github/workflows/githubci.yml
@@ -7,7 +7,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        arduino-platform: ['metro_m0', 'metro_m4', 'nrf52840', 'esp8266', 'esp32']
+        arduino-platform: ['metro_m0', 'metro_m4', 'nrf52840']
 
     runs-on: ubuntu-latest
     steps:

--- a/src/Adafruit_SPIFlashBase.cpp
+++ b/src/Adafruit_SPIFlashBase.cpp
@@ -179,9 +179,9 @@ bool Adafruit_SPIFlashBase::begin(SPIFlash_Device_t const *flash_devs,
     addr_byte = 4;
     // Enable 4-Byte address mode (This has to be done after the reset above)
     _trans->runCommand(SFLASH_CMD_4_BYTE_ADDR);
-  }else if (_flash_dev->total_size > 64 * 1024) {
+  } else if (_flash_dev->total_size > 64 * 1024) {
     addr_byte = 3;
-  }else {
+  } else {
     addr_byte = 2;
   }
 

--- a/src/Adafruit_SPIFlashBase.cpp
+++ b/src/Adafruit_SPIFlashBase.cpp
@@ -175,15 +175,14 @@ bool Adafruit_SPIFlashBase::begin(SPIFlash_Device_t const *flash_devs,
 
   // Addressing byte depends on total size
   uint8_t addr_byte;
-  if (_flash_dev->total_size < 64 * 1024) {
-    addr_byte = 2;
-  } else if (_flash_dev->total_size < 16 * 1024 * 1024) {
-    addr_byte = 3;
-  } else {
+  if (_flash_dev->total_size > 16 * 1024 * 1024) {
     addr_byte = 4;
-
     // Enable 4-Byte address mode (This has to be done after the reset above)
     _trans->runCommand(SFLASH_CMD_4_BYTE_ADDR);
+  }else if (_flash_dev->total_size > 64 * 1024) {
+    addr_byte = 3;
+  }else {
+    addr_byte = 2;
   }
 
   _trans->setAddressLength(addr_byte);

--- a/src/spi/Adafruit_FlashTransport_SPI.cpp
+++ b/src/spi/Adafruit_FlashTransport_SPI.cpp
@@ -23,7 +23,6 @@
  */
 
 #include "Adafruit_FlashTransport.h"
-#include "Arduino.h"
 
 Adafruit_FlashTransport_SPI::Adafruit_FlashTransport_SPI(
     uint8_t ss, SPIClass *spiinterface) {
@@ -107,11 +106,13 @@ void Adafruit_FlashTransport_SPI::fillAddress(uint8_t *buf, uint32_t addr) {
   switch (_addr_len) {
   case 4:
     *buf++ = (addr >> 24) & 0xFF;
-    __attribute__((fallthrough));
+    //__attribute__((fallthrough)); ESP32 doesn't support this attribute yet
+    // fall through
 
   case 3:
     *buf++ = (addr >> 16) & 0xFF;
-    __attribute__((fallthrough));
+    //__attribute__((fallthrough)); ESP32 doesn't support this attribute yet
+    // fall through
 
   case 2:
   default:

--- a/src/spi/Adafruit_FlashTransport_SPI.h
+++ b/src/spi/Adafruit_FlashTransport_SPI.h
@@ -25,6 +25,7 @@
 #ifndef ADAFRUIT_FLASHTRANSPORT_SPI_H_
 #define ADAFRUIT_FLASHTRANSPORT_SPI_H_
 
+#include "Arduino.h"
 #include "SPI.h"
 
 class Adafruit_FlashTransport_SPI : public Adafruit_FlashTransport {

